storage/fanout.go

```diff
@@ -350,13 +351,26 @@ func (c *mergeSeriesSet) At() Series {
 
 func (c *mergeSeriesSet) Err() error {
 	for _, set := range c.sets {
+		var err error
 		if err := set.Err(); err != nil {
-			return err
+			switch eType := reflect.TypeOf(set).String(); eType {
+			case "tsdb.errSeriesSet":
+				return err
+			case "remote.errSeriesSet":
+				err = RemoteStorageError("tsdb error")
+			}
```

Comment:
a big drawback I see is that if `tsdb.errSeriesSet` gets refactoried and the struct has a different name it will brake the code. 
or if I change tsdb.errSeriesSet  to be exported (tsdb.ErrSeriesSet ) so than can use 
```
switch set.(type) {
   case tsdb.ErrSeriesSet:
....
```
and the compiler will complain if the `type ErrSeriesSet` is no longer valid.